pkg/image/validation/elemental.go

```diff
@@ -0,0 +1,72 @@
+package validation
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+)
+
+const (
+	elementalComponent      = "Elemental"
+	elementalConfigFilename = "elemental_config.yaml"
+)
+
+func validateElemental(ctx *image.Context) []FailedValidation {
+	var failures []FailedValidation
+
+	elementalConfigDir := filepath.Join(ctx.ImageConfigDir, "elemental")
+	if _, err := os.Stat(elementalConfigDir); err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+
+		failures = append(failures, FailedValidation{
+			UserMessage: "Elemental config directory could not be read",
+			Error:       err,
+		})
+		return failures
+	}
+
+	failures = append(failures, validateElementalDir(elementalConfigDir)...)
+
+	if ctx.ImageDefinition.OperatingSystem.Packages.RegCode == "" {
+		failures = append(failures, FailedValidation{
+			UserMessage: "Operating system package registration code field must be defined when using Elemental with SL Micro 6.0",
+		})
+	}
+
+	return failures
+}
+
+func validateElementalDir(elementalConfigDir string) []FailedValidation {
+	var failures []FailedValidation
+
+	elementalConfigDirEntries, err := os.ReadDir(elementalConfigDir)
+	if err != nil {
+		failures = append(failures, FailedValidation{
```

Comment:
We need to return after this. No point in trying to make assertions on what's in the directory if we're unable to read it.